iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -81,69 +134,51 @@ class Sample
     operator bool() const;
 
     ///
-    /// @brief allocation Access to the memory chunk loaned to the sample.
-    /// @return
+    /// @brief allocation Access to the encapsulated type loaned to the sample.
+    /// @return a pointer to the encapsulated type.
+    /// @details Only available for non-const type T.
     ///
+    template <typename S = T, typename = std::enable_if_t<std::is_same<S, T>::value && !std::is_const<S>::value, S>>
     T* get() noexcept;
 
     ///
-    /// @brief allocation Read-only access to the memory chunk loaned to the sample.
-    /// @return
+    /// @brief allocation Read-only access to the encapsulated type loaned to the sample.
+    /// @return a const pointer to the encapsulated type.
     ///
     const T* get() const noexcept;
 
     ///
     /// @brief header Retrieve the header of the underlying memory chunk loaned to the sample.
     /// @return The ChunkHeader of the underlying memory chunk.
+    /// @details Only available for non-const type T.
     ///
+    template <typename S = T, typename = std::enable_if_t<std::is_same<S, T>::value && !std::is_const<S>::value, S>>
     mepoo::ChunkHeader* getHeader() noexcept;
 
+    ///
+    /// @brief header Retrieve the header of the underlying memory chunk loaned to the sample.
+    /// @return The const ChunkHeader of the underlying memory chunk.
+    ///
+    const mepoo::ChunkHeader* getHeader() const noexcept;
```

Comment:
Theoretically a pointer can always be a nullptr. To promise the user that they always get a pointer to a valid header I would return here a reference like `const mepoo::ChunkHeader& getHeader() const noexcept;` Otherwise the question always arises, do I have to check that the pointer != nullptr when I access it.

Reply:
will be done in a follow up PR, like the non-const variant